sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs

```diff
@@ -367,12 +367,27 @@ public async Task RecognizeHealthcareEntitiesBatchWithCancellation()
                 batchDocuments.Add(document);
             }
 
-            AnalyzeHealthcareEntitiesOperation operation = await client.StartAnalyzeHealthcareEntitiesAsync(batchDocuments, "en");
+            AnalyzeHealthcareEntitiesOperation operation = default;
 
-            await operation.CancelAsync();
+            await TestRetryHelper.RetryAsync(async () =>
+            {
+                try
+                {
+                    operation = await client.StartAnalyzeHealthcareEntitiesAsync(batchDocuments, "en");
+                    await operation.CancelAsync();
+                    await operation.WaitForCompletionAsync();
+                }
+                catch (Exception e)
+                {
+                    Assert.AreEqual(typeof(RequestFailedException), e.GetType());
+                    Assert.IsTrue(e.Message.Contains("The operation was canceled so no value is available."));
+                    return (Response)null;
+                }
 
-            RequestFailedException ex = Assert.ThrowsAsync<RequestFailedException>(async () => await operation.WaitForCompletionAsync());
-            Assert.IsTrue(ex.Message.Contains("The operation was canceled so no value is available."));
+                // If we get here, that means that the operation completed successfully and didn't cancel.
+                throw new InvalidOperationException("StartAnalyzeHealthcareEntitiesAsync did not cancel operation");
+            },
+            maxIterations: 15, delay: TimeSpan.FromSeconds(0));
```

Comment:
should there be a wait time added just in case? maybe 1 second?

Reply:
I added a 1 second delay just in case repeated requests cause an issue with the service. Otherwise I don't think it would impact the semantics of the test itself since our issue is that the operation completes successfully before getting cancelled.